readthedocs/api/v2/views/integrations.py

```diff
@@ -290,19 +356,21 @@ def handle_webhook(self):
         if event in (GITHUB_CREATE, GITHUB_DELETE):
             return self.sync_versions(self.project)
 
-        if (
-            event == GITHUB_PULL_REQUEST and
-            self.data['action'] in [GITHUB_PULL_REQUEST_OPEN, GITHUB_PULL_REQUEST_SYNC]
-        ):
-            try:
-                identifier, verbose_name = self.get_external_version_data()
-                external_version = get_or_create_external_version(
-                    self.project, identifier, verbose_name
-                )
-                return self.get_response_push(self.project, [external_version.verbose_name])
-
-            except KeyError:
-                raise ParseError('Parameters "sha" and "number" are required')
+        if event == GITHUB_PULL_REQUEST and action:
+            if (
+                action in
+                [
+                    GITHUB_PULL_REQUEST_OPENED,
+                    GITHUB_PULL_REQUEST_REOPENED,
+                    GITHUB_PULL_REQUEST_SYNC
+                ]
+            ):
+                # Handle opened, synchronize, reopened pull_request event.
+                return self.get_external_version_response(self.project)
+
+            if action == GITHUB_PULL_REQUEST_CLOSED:
+                # Handle closed pull_request event.
+                return self.delete_external_version(self.project)
 
         return None
```

Comment:
This was here because 
https://github.com/rtfd/readthedocs.org/blob/fb793cad743aa3c849e67e6ddf686625d920b557/readthedocs/api/v2/views/integrations.py#L80-L84